math-scala/src/main/scala/org/apache/mahout/math/scalabindings/RLikeMatrixOps.scala

```diff
@@ -17,14 +17,22 @@
 package org.apache.mahout.math.scalabindings
 
 import org.apache.mahout.math.function.Functions
-import org.apache.mahout.math.{Vector, Matrix}
+import org.apache.mahout.math.{Matrix, Vector}
 import scala.collection.JavaConversions._
 import RLikeOps._
+import org.apache.mahout.math.scalabindings._
+
 
 class RLikeMatrixOps(m: Matrix) extends MatrixOps(m) {
 
   /** Structure-optimized mmul */
-  def %*%(that: Matrix) = MMul(m, that, None)
+
+  implicit var solverOperator: opMMulSolver =_
```

Comment:
Somehow i don't like implicit vars inside the decorator pattern. Clean decorator pattern (which is what it really is) assumes there's no new intrinsic state. so the only value instrinsic is `m`.

Instead, it should query solvers from a table as instantiated and saved in the current solver factory.